plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/storage/ScaleIOStoragePoolTest.java

```diff
@@ -140,55 +135,173 @@ public void testDefaults() {
+            when(ScaleIOUtil.getSystemIdForVolume(volumeId)).thenReturn(systemId);
+
+            KVMPhysicalDisk disk = adapter.getPhysicalDisk(volumePath, pool);
+            assertNotNull(disk);
+            assertEquals("/dev/disk/by-id/emc-vol-" + systemId + "-6c3362b500000001", disk.getPath());
+            assertEquals(PhysicalDiskFormat.RAW, disk.getFormat());
+            assertEquals(8192, disk.getSize());
+            assertEquals(8192, disk.getVirtualSize());
+        }
+    }
+
+    @Test
+    public void testGetPhysicalDiskWithSystemIdFileExistsWithDetails() {
+        when(Script.runSimpleBashScript("/opt/emc/scaleio/sdc/bin/drv_cfg --query_mdms|grep " + systemId + "|awk '{print $5}'")).thenReturn(sdcId);
+        ScaleIOStoragePool pool = new ScaleIOStoragePool(uuid, "192.168.1.19", 443, "a519be2f00000000", type, details, adapter);
+        final String volumePath = "6c3362b500000001:vol-139-3d2c-12f0";
+        final String volumeId = "6c3362b500000001";
+        OutputInterpreter.OneLineParser spyParser = spy(new OutputInterpreter.OneLineParser());
+
+        try (MockedConstruction<QemuImg> ignored = Mockito.mockConstruction(QemuImg.class, (mockQemuImg, contextQemuImg) -> {
+            Map<String, String> details = new HashMap<>();
+            details.put(QemuImg.VIRTUAL_SIZE, "16384");
+            details.put(QemuImg.FILE_FORMAT, PhysicalDiskFormat.QCOW2.toString());
+            Mockito.doReturn(details).when(mockQemuImg).info(any(QemuImgFile.class));
+        }); MockedConstruction<File> ignored1 = Mockito.mockConstruction(File.class, (mockFile, contextFile) -> {
+            Mockito.when(mockFile.exists()).thenReturn(true);
+        }); MockedConstruction<Script> ignored2 = Mockito.mockConstruction(Script.class, (mockScript, contextScript) -> {
+            Mockito.doReturn(null).when(mockScript).execute(any(OutputInterpreter.class));
+        }); MockedConstruction<OutputInterpreter.OneLineParser> ignored3 = Mockito.mockConstruction(OutputInterpreter.OneLineParser.class, withSettings().spiedInstance(spyParser), (mockParser, contextParser) -> {
+            Mockito.doReturn("16384").when(mockParser).getLine();
+        }); MockedStatic<ScaleIOUtil> ignored5 = Mockito.mockStatic(ScaleIOUtil.class, Mockito.CALLS_REAL_METHODS)) {
+            when(ScaleIOUtil.getSystemIdForVolume(volumeId)).thenReturn(systemId);
+
+            KVMPhysicalDisk disk = adapter.getPhysicalDisk(volumePath, pool);
+            assertNotNull(disk);
+            assertEquals("/dev/disk/by-id/emc-vol-" + systemId + "-6c3362b500000001", disk.getPath());
+            assertEquals(PhysicalDiskFormat.QCOW2, disk.getFormat());
+            assertEquals(16384, disk.getSize());
+            assertEquals(16384, disk.getVirtualSize());
+        }
+    }
+
     @Test
     public void testConnectPhysicalDisk() {
+        when(Script.runSimpleBashScript("/opt/emc/scaleio/sdc/bin/drv_cfg --query_mdms|grep " + systemId + "|awk '{print $5}'")).thenReturn(sdcId);
```

Comment:
This string is getting duplicated a lot. Can we replace this with something like this?
```java

DRV_CFG_SCRIPT = "/opt/emc/scaleio/sdc/bin/drv_cfg --query_mdms|grep %s |awk '{print $5}'"
...
when(Script.runSimpleBashScript(String.format(DRV_CFG_SCRIPT, systemId)).thenReturn(sdcId);
```